src/WebJobs.Script/Description/Script/ScriptFunctionInvoker.cs

```diff
@@ -77,59 +76,43 @@ public static bool IsSupportedScriptType(ScriptType scriptType)
+            string functionInstanceOutputPath = Path.Combine(Path.GetTempPath(), "Functions", "Binding", invocationId);
 
-                object convertedInput = ConvertInput(input);
-                Utility.ApplyBindingData(convertedInput, binder.BindingData);
-                Dictionary<string, object> bindingData = binder.BindingData;
-                bindingData["InvocationId"] = invocationId;
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            InitializeEnvironmentVariables(environmentVariables, functionInstanceOutputPath, input, _outputBindings, functionExecutionContext);
 
-                await ProcessInputBindingsAsync(convertedInput, functionInstanceOutputPath, binder, _inputBindings, _outputBindings, bindingData, environmentVariables);
+            object convertedInput = ConvertInput(input);
+            Utility.ApplyBindingData(convertedInput, binder.BindingData);
+            Dictionary<string, object> bindingData = binder.BindingData;
+            bindingData["InvocationId"] = invocationId;
 
-                // TODO
-                // - put a timeout on how long we wait?
-                // - need to periodically flush the standard out to the TraceWriter
-                Process process = CreateProcess(path, workingDirectory, arguments, environmentVariables);
-                process.Start();
-                process.WaitForExit();
+            await ProcessInputBindingsAsync(convertedInput, functionInstanceOutputPath, binder, _inputBindings, _outputBindings, bindingData, environmentVariables);
 
-                string output = process.StandardOutput.ReadToEnd();
-                TraceWriter.Info(output);
-                traceWriter.Info(output);
-
-                startedEvent.Success = process.ExitCode == 0;
-
-                if (!startedEvent.Success)
+            Process process = CreateProcess(path, workingDirectory, arguments, environmentVariables);
+            process.OutputDataReceived += (s, e) =>
+            {
+                if (e.Data != null)
                 {
-                    string error = process.StandardError.ReadToEnd();
-                    throw new ApplicationException(error);
+                    TraceWriter.Info(e.Data);
+                    traceWriter.Info(e.Data);
                 }
+            };
 
-                await ProcessOutputBindingsAsync(functionInstanceOutputPath, _outputBindings, input, binder, bindingData);
+            process.Start();
+            process.BeginOutputReadLine();
 
-                TraceWriter.Info(string.Format("Function completed (Success, Id={0})", invocationId));
-            }
-            catch
-            {
-                startedEvent.Success = false;
-                TraceWriter.Error(string.Format("Function completed (Failure, Id={0})", invocationId));
-                throw;
-            }
-            finally
+            await Task.Run(() => process.WaitForExit());
```

Comment:
I actually did this first -- it was a bunch more code so I went with the simpler version :-)

I can switch it back.

---
In reply to: [75989020](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r75989020) [](ancestors = 75989020)